File: ios/WalletCore/Error/CError.swift
//
//  CError.swift
//  trust-wallet-core-react-native
//
//  Created by Nguyen Van Toan on 21/5/24.
//

import Foundation

enum CError : Error {
    case MissingArgumentError
    case InvalidHexArgumentError
    case UnSupportedCoin
    case InvalidMnemonic
    case Unknown
    case UnsupportedTransaction
}

extension CError {
    func toReactException() -> String{
        var error = ""
        
        switch self {
        case .MissingArgumentError:
            error = "Missing argument error"
            break
        case .InvalidHexArgumentError:
            error = "Invalid hex argument error"
        case .UnSupportedCoin:
            error = "Unsupported coin"
            break
        case .InvalidMnemonic:
            error = "Invalid mnemonic"
            break
        case .Unknown:
            error = "Unknown error"
            break
        case .UnsupportedTransaction:
            error = "UnSupported Transaction"
            break
        }
        
        return error
    }
    
    func toReactCode() -> String {
        var code = ""
        
        switch self {
        case .MissingArgumentError:
            code = "MissingArgumentError"
            break
        case .InvalidHexArgumentError:
            code = "InvalidHexArgumentError"
            break
        case .UnSupportedCoin:
            code = "UnSupportedCoin"
            break
        case .InvalidMnemonic:
            code = "InvalidMnemonic"
            break
        case .Unknown:
            code = "Unknown"
            break
        case .UnsupportedTransaction:
            code = "UnsupportedTransaction"
            break
        }
        
        return code
    }
}

File: android/src/main/java/com/trustwalletcorereactnative/wallet_core/error/CError.kt
package com.trustwalletcorereactnative.wallet_core.error

enum class CEError(val message : String,val code :String){
    MissingArgumentError("Missing argument error", "MissingArgumentError"),
    UnSupportedCoin("Unsupported coin","UnSupportedCoin"),
    InvalidMnemonic("Invalid mnemonic","InvalidMnemonic"),
    Unknown("Unknown error","Unknown"),
    UnsupportedTransaction("UnSupported Transaction","UnsupportedTransaction")
}


final class CError : Exception{
    private var code : String

    constructor(message: String,code: String): super(message){
        this.code = code
    }

    public fun getCode() : String{
        return code
    }
}